core/dnsserver/config.go

```diff
@@ -60,8 +70,14 @@ type Config struct {
 	// firstConfigInBlock is used to reference the first config in a server block, for the
 	// purpose of sharing single instance of each plugin among all zones in a server block.
 	firstConfigInBlock *Config
+
+	// metaCollector references the first MetadataCollector plugin, if one exists
+	metaCollector *MetadataCollector
```

Comment:
I think we do not have to declare it as a pointer as it is an Interface type that is already a pointer to the real structure
```suggestion
	metaCollector MetadataCollector
```